File: CultureLinkCRM/CultureLinkCRM.Web/Modules/CrmDB/Address/AddressColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace CultureLinkCRM.CrmDB.Columns;

[ColumnsScript("CrmDB.Address")]
[BasedOnRow(typeof(AddressRow), CheckNames = true)]
public class AddressColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int AddressId { get; set; }

    [EditLink, DisplayName("Address")]
    public string AddressLine1 { get; set; }    

    public string CityName { get; set; }

    //public string JurisdictionName { get; set; }

    public string JurisdictionAbbreviation { get; set; }    

    public string ZipCodeName { get; set; }
}

File: CultureLinkCRM/CultureLinkCRM.Web/Modules/ServerTypes/CrmDB/AddressColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { AddressRow } from "./AddressRow";

export interface AddressColumns {
    AddressId: Column<AddressRow>;
    AddressLine1: Column<AddressRow>;
    CityName: Column<AddressRow>;
    JurisdictionAbbreviation: Column<AddressRow>;
    ZipCodeName: Column<AddressRow>;
}

export class AddressColumns extends ColumnsBase<AddressRow> {
    static readonly columnsKey = 'CrmDB.Address';
    static readonly Fields = fieldsProxy<AddressColumns>();
}